src/solvers/refinement/string_refinement.cpp

```diff
@@ -105,12 +105,11 @@ void string_refinementt::add_instantiations()
       debug() << from_expr(ns, "", j) << "; ";
     debug() << "}"  << eom;
 
-    for(const auto &j : i.second)
+    for(const auto &ua : universal_axioms)
     {
-      const exprt &val=j;
-
-      for(const auto &ua : universal_axioms)
+      for(const auto &j : i.second)
       {
+        const exprt &val=j;
```

Comment:
Is this line strictly necessary, or could we pass `j` directly to `instantiate` (or rename `j` to `val`)?